newa/__init__.py

```diff
@@ -742,7 +747,219 @@ class IssueAction:  # type: ignore[no-untyped-def]
+        """
+
+        if assignee_email not in self.user_names:
+            assignee_names = [u.name for u in self.connection.search_users(user=assignee_email)]
+            if not assignee_names or len(assignee_names) > 1:
+                raise Exception(f"Exactly one Jira user is expected to match {assignee_email}"
+                                f"({', '.join(assignee_names)})!")
+            self.user_names[assignee_email] = assignee_names[0]
+
+        return self.user_names[assignee_email]
+
+    def get_details(self, issue: Issue) -> jira.Issue:
+        """ Return issue details """
+
+        try:
+            return self.connection.issue(issue.id)
+        except jira.JIRAError as e:
+            raise Exception(f"Jira issue {issue} not found!") from e
+
+    def get_open_issues(self,
+                        action: IssueAction,
+                        all_respins: bool = False) -> dict[str, dict[str, str]]:
+        """
+        Get issues related to erratum job with given summary
+
+        Unless 'all_respins' is defined only issues related to the current respin are returned.
+        Result is a dictionary such that keys are found Jira issue keys (ID) and values
+        are dictionaries such that there is always 'description' key and if the issues has
+        parent then there is also 'parent' key. For instance:
+
+        {
+            "NEWA-123": {
+                "description": "description of first issue",
+                "parent": "NEWA-456"
+            }
+            "NEWA-456": {
+                "description": "description of second issue"
+            }
+        }
+        """
+
+        fields = ["description", "parent"]
+
+        newa_description = f"{self.newa_id(action, True) if all_respins else self.newa_id(action)}"
+        query = \
+            f"project = '{self.project}' AND " + \
+            f"labels in ({IssueHandler.newa_label}) AND " + \
+            f"description ~ '{newa_description}' AND " + \
+            f"status not in ({','.join(self.transitions['closed'])})"
+
+        search_result = self.connection.search_issues(query, fields=fields, json_result=True)
+        if not isinstance(search_result, dict):
+            raise Exception(f"Unexpected search result type {type(search_result)}!")
+
+        # Transformation of search_result json into simpler structure gets rid of
+        # linter warning and also makes easier mocking (for tests).
+        result = {}
+        for jira_issue in search_result["issues"]:
+            result[jira_issue["key"]] = {"description": jira_issue["fields"]["description"]}
+            if "parent" in jira_issue["fields"]:
+                result[jira_issue["key"]] |= {"parent": jira_issue["fields"]["parent"]["key"]}
+        return result
+
+    def create_issue(self,
+                     action: IssueAction,
+                     summary: str,
+                     description: str,
+                     assignee_email: str,
```

Comment:
Can  we make assignee optional? In Jira it is quite common to have issues unassigned, however current implementation requires it.
```
Traceback (most recent call last):
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/bin/newa", line 8, in <module>
    sys.exit(main())
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/core.py", line 1719, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/ksrot/.local/share/hatch/env/virtual/newa/j_Eh7vfQ/dev/lib/python3.9/site-packages/click/decorators.py", line 45, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/ksrot/devel/github/newa/newa/cli.py", line 220, in cmd_jira
    config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
  File "/home/ksrot/devel/github/newa/newa/__init__.py", line 312, in from_yaml_file
    return cls.from_yaml(filepath.read_text())
  File "/home/ksrot/devel/github/newa/newa/__init__.py", line 308, in from_yaml
    return cls(**data)
  File "<attrs generated init newa.ErratumConfig>", line 6, in __init__
  File "/home/ksrot/devel/github/newa/newa/__init__.py", line 755, in <lambda>
    factory=list, converter=lambda issues: [
  File "/home/ksrot/devel/github/newa/newa/__init__.py", line 756, in <listcomp>
    IssueAction(**issue) for issue in issues])
TypeError: __init__() missing 1 required positional argument: 'assignee'
```

Reply:
Fixed (and tested).